lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -400,8 +400,7 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
         }
 
         SmallVector<char, 64> Buf;
-        // TODO: Handle special names
-        StringRef Name = (VD->getBaseName().getIdentifier().str() +
+        StringRef Name = (VD->getBaseName().userFacingName() +
                           Twine(Kind)).toStringRef(Buf);
```

Comment:
@adrian-prantl, this is the no-functionality-change implementation, and I think it does make sense to land that first, but it's worth considering whether this makes sense to begin with. Subscripts can be overloaded.